examples/sinh_libtorch/CMakeLists.txt

```diff
@@ -16,3 +16,8 @@ target_link_libraries(sinh_example "-Wl,-no-as-needed ${TORCH_INSTALL_PREFIX}/li
 
 target_link_libraries(sinh_example ${TORCH_LIBRARIES})
 target_link_libraries(sinh_example nvfuser_codegen)
+target_include_directories(sinh_example SYSTEM PRIVATE
+  ../../csrc
```

Comment:
Definitely, the correct solution is to add a new exported variable to our generated+installed CMake config file, and then picking up on that for the build of this project.

It's not the source files that are needed, just the headers. So no need to change the package.

I think the problem here is that we never set a `target_include_directories(nvfuser PUBLIC ${CMAKE_SOURCE_DIR}/csrc)`, so it doesn't know that the csrc header files are a public dependency of nvFuser as opposed to a private one. That is, I think CMake should be automagically creating + using your hypothetical `NVFUSER_INCLUDE_DIRECTORIES` variable.

Note also that we're really making way too many headers public, and similarly packaging way too many headers. As per my earlier comment though, I want to step back from boiling the ocean here ;-)

Looking at this again I also see that I've `set_target_properties` *twice* on `nvf_py_internal`, which is harmless but confusing. I'll fix that too.

- [ ] Add `${CMAKE_SOURCE_DIR}/csrc` as a `PUBLIC` include dir for the `nvfuser` target, and check to make that that this means the proper dir is included for `sinh_libtorch`.
- [ ] Don't `set_target_properties` on `nvf_py_internal` twice.

Reply:
I was clearly confused when writing the above. The exact thing I said doesn't work, because it doesn't make sense ;-)

This isn't needed. the `target_link_libraries(sinh_example nvfuser_codegen)` will implicitly pull in the appropriate directories, because we:

```cmake
target_include_directories(codegen_internal PUBLIC
  "$<BUILD_INTERFACE:${NVFUSER_SRCS_DIR}>"
  "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/nvfuser>"
)
```

in the main cmake.

I think my confusion was thinking that we needed to link the `nvfuser` library instead of the `nvfuser_codegen` library.

Anyway, just going to remove this part. Thanks for calling it to my attention, Xiao!